design/UsageValidationRules.md

```diff
@@ -0,0 +1,71 @@
+## Definition of "resources being used"
+
+Resources in WebGPU can be used in different ways that are declared at resource creation time and interact with various validation rules.
+
+Buffers have the following usages (and commands inducing that usage):
+ - `VERTEX` for the `buffers` in calls to `WebGPUCommandEncoder.setVertexBuffers`
+ - `INDEX` for `buffer` in calls to `WebGPUCommandEncoder.setIndexBuffer`
+ - `INDIRECT` for `indirectBuffer` in calls to `WebGPUCommandEncode.{draw|drawIndexed|dispatch}Indirect`
+ - `UBO`, `RO_STORAGE` and `STORAGE` for buffers referenced by bindgroups passed to `setBindGroup`, with the usage corresponding to the binding's type.
+ - `TRANSFER_SRC` for buffers used as the copy source of various commands.
+ - `TRANSFER_DST` for buffers used as the copy destination of various commands.
+ - (Maybe [RO_]STORAGE_TEXEL and UNIFORM_TEXEL?)
```

Comment:
yeah, I believe viewing buffers in terms of texels is supported on all next-gen APIs rather uniformly, and we should have the usage defined here

Reply:
AFAIK, D3D does not support treating buffers as texture views that you can sample from.

Reply:
In D3D both textures and buffers are seen by the shader as views, and in this case SRV can be created from a buffer with [CreateShaderResourceView](https://msdn.microsoft.com/en-us/library/windows/desktop/dn788672(v=vs.85).aspx). Am I missing something?

Reply:
We implemented texel buffers as structured buffers. Texel buffer only allow load/fetch (no sampling).

Reply:
My apologies, @kvark and @msiglreith .  I stand corrected!